src/auxkernels/MFEMGradAux.C

```diff
@@ -0,0 +1,41 @@
+#pragma once
+#include "MFEMGradAux.h"
+
+registerMooseObject("PlatypusApp", MFEMGradAux);
+
+/*
+Class to set an H(curl) auxvariable to be the gradient of a H1 scalar variable.
+*/
+InputParameters
+MFEMGradAux::validParams()
+{
+  InputParameters params = MFEMAuxKernel::validParams();
+  params.addClassDescription(
+      "Calculates the gradient of a H1 conforming source variable and stores the result"
+      " on a H(curl) conforming ND result auxvariable");
+  params.addRequiredParam<VariableName>("source",
+                                        "Scalar H1 MFEMVariable to take the gradient of.");
+  params.addParam<mfem::real_t>("scale_factor", 1.0, "Factor to scale result auxvariable by.");
+  return params;
+}
+
+MFEMGradAux::MFEMGradAux(const InputParameters & parameters)
+  : MFEMAuxKernel(parameters),
+    _source_var_name(getParam<VariableName>("source")),
+    _source_var(*getMFEMProblem().getProblemData()._gridfunctions.Get(_source_var_name)),
+    _scale_factor(getParam<mfem::real_t>("scale_factor")),
+    _h1_fespace(*_source_var.ParFESpace()),
+    _hcurl_fespace(*_result_var.ParFESpace()),
```

Comment:
Do these actually need to be stored as class members? Can't you just access them to construct the curl operator, without saving references to them?